crates/common_utils/src/validation.rs

```diff
@@ -57,6 +60,24 @@ pub fn validate_email(email: &str) -> CustomResult<(), ValidationError> {
     Ok(())
 }
 
+/// Checks whether a given domain matches against a list of valid domain glob patterns
+pub fn validate_domain_against_allowed_domains(
+    domain: &str,
+    allowed_domains: HashSet<String>,
+) -> bool {
+    allowed_domains.iter().any(|allowed_domain| {
+        Glob::new(allowed_domain)
+            .map(|glob| glob.compile_matcher().is_match(domain))
+            .map_err(|err| {
+                let err_msg = format!("Invalid glob pattern! - {:?}", err);
```

Comment:
can we write a more descriptive error message which states failed for which flow etc.

Reply:
flows etc, should be available in the logs context. We can add payment specific context if needed